resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala

```diff
@@ -462,6 +462,30 @@ package object config extends Logging {
     .stringConf
     .createWithDefault("yarn.io/fpga")
 
+  private[spark] val YARN_CLIENT_STAT_CACHE_PRELOAD_ENABLED =
+    ConfigBuilder("spark.yarn.client.statCache.preload.enabled")
+    .doc("This configuration enables statCache to be preloaded at YARN client side. This feature " +
+      "analyzes the pattern of resources paths, and if multiple resources shared the same parent " +
+      "directory, a single <code>listStatus</code> will be invoked on the parent directory " +
+      "instead of multiple <code>getFileStatus</code> performed on each individual resources. " +
+      "If most resources from a small set of directories, this can substantially improve job " +
+      "submission time. Enabling this feature may potentially increase client memory overhead.")
```

Comment:
Do you mean `listStatus ` with `PathFilter `? I can try that. 

https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/fs/FileSystem.html#listStatus-org.apache.hadoop.fs.Path-org.apache.hadoop.fs.PathFilter-

Reply:
Optimized with `PathFilter`. Validate internally. Now HDFS will only return needed fileStatus instead of all fileStatus from directory. This will limit the memory consumption.